### PR TITLE
fix: initialize BatchNorm2d buffers only on meta

### PR DESCRIPTION
**Pull request by tarekziade**
# What does this PR do?

Commit 8dd9c999a6262d6ceb48f4a2da7acaccfa80e3bc introduced a regression by unconditionally reinitializing BatchNorm2d buffers (running_mean, running_var, num_batches_tracked) in the _init_weights() method.

The problem: When loading pretrained timm models, the flow is:
1. timm.create_model(pretrained=True) correctly loads pretrained BatchNorm statistics
2. post_init() calls _init_weights() on all modules
3. _init_weights() overwrites the pretrained BatchNorm buffers with zeros/ones
4. Model produces incorrect results because normalization uses wrong statistics

The error shows a mismatch at index (0, 178, 0, 0) with a difference of 80224.0 on CircleCI on AVX 512 CPUs

The fix: added a check to only initialize BatchNorm2d buffers when they're on the meta device (uninitialized): This ensures pretrained buffers are preserved while still handling the meta device initialization case.

### CODE REVIEW

File: src/transformers/models/timm_backbone/modeling_timm_backbone.py [hunk 1]
```diff
@@ -117,10 +117,26 @@ def _init_weights(self, module):
...
         if hasattr(module, "init_non_persistent_buffers"):
             module.init_non_persistent_buffers()
-        elif isinstance(module, nn.BatchNorm2d) and getattr(module, "running_mean", None) is not None:
-            init.zeros_(module.running_mean)
-            init.ones_(module.running_var)
-            init.zeros_(module.num_batches_tracked)
+        elif isinstance(module, nn.BatchNorm2d):
+            running_mean = getattr(module, "running_mean", None)
+            running_var = getattr(module, "running_var", None)
+            num_batches_tracked = getattr(module, "num_batches_tracked", None)
```
Comment (Cyrilvallez) [inline src/transformers/models/timm_backbone/modeling_timm_backbone.py]: I one exists, all are guaranteed to exist, so we can only check one (see https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/batchnorm.py#L61-L79)

File: src/transformers/models/timm_backbone/modeling_timm_backbone.py [hunk 2]
```diff
@@ -117,10 +117,26 @@ def _init_weights(self, module):
...
+            if (
+                running_mean is not None
+                and running_var is not None
+                and (
+                    running_mean.device.type == "meta"
```
Comment (Cyrilvallez) [inline src/transformers/models/timm_backbone/modeling_timm_backbone.py]: We can simply check meta device here I believe! Should not need to check for limit values

**Review by Cyrilvallez — APPROVED** [review]
Ha indeed, very nice thanks! We can simplify a bit the different checks though!

**Comment by tarekziade** [comment]
run-slow: timm_backbone, timm_wrapper
